raft/src/main/java/org/apache/kafka/raft/internals/RecordsIterator.java

```diff
@@ -180,6 +184,11 @@ private Optional<Batch<T>> nextBatch() {
     }
 
     private Batch<T> readBatch(DefaultRecordBatch batch) {
+        if (doCrcValidation) {
+            // Perform a CRC validity check on this batch
+            batch.ensureValid();
```

Comment:
This throws an exception that doesn't include a stacktrace. This is a rare event so we want to pay the cost of generating a stacktrace as it will help us debug the issue.